src/web/web-service/mdns_service.hpp

```diff
@@ -28,67 +28,87 @@
 
 /**
  * @file
- *   This file implements the web service of border router
+ *   This file implements the mdns service
  */
 
-#ifndef WEB_SERVICE
-#define WEB_SERVICE
+#ifndef MNDS_SERVICE
+#define MNDS_SERVICE
 
-#include <algorithm>
 #include <fstream>
 #include <iostream>
 #include <string>
+#include <thread>
 #include <vector>
 
 #include <net/if.h>
 
-#include <boost/property_tree/ptree_fwd.hpp>
-#include <boost/asio/ip/tcp.hpp>
+#include <jsoncpp/json/json.h>
+#include <jsoncpp/json/writer.h>
 
+#include "../mdns-publisher/mdns_publisher.hpp"
 #include "../wpan-controller/wpan_controller.hpp"
+#include "common/logging.hpp"
 
-namespace SimpleWeb {
-template<class T> class Server;
-typedef boost::asio::ip::tcp::socket HTTP;
-}
+#define OT_BORDER_ROUTER_PORT 49191
 
 namespace ot {
 namespace Web {
 
-typedef SimpleWeb::Server<SimpleWeb::HTTP> HttpServer;
-
-class WebServer
+/**
+ * This class provides web service to multicast DNS.
+ *
+ */
+class MdnsService
 {
 public:
-    WebServer(void);
-    ~WebServer(void);
-    void StartWebServer(const char *aIfName);
+
+    /**
+     * This method handles http request to advertise service.
+     *
+     * @returns The string to the http response of mdns service.
+     *
+     */
+    std::string HandleMdnsRequest(const std::string &aMdnsRequest);
+
+    /**
+     * This method starts mdns service.
+     *
+     * @param[in]  aNetworkName  The reference to the network name.
+     * @param[in]  aExtPanId     The reference to the extend PAN ID.
+     *
+     */
+    int StartMdnsService(const std::string &aNetworkName, const std::string &aExtPanId);
+
+    /**
+     * This method updates mdns service.
+     *
+     * @param[in]  aNetworkName  The reference to the network name.
+     * @param[in]  aExtPanId     The reference to the extend PAN ID.
+     *
+     */
+    int UpdateMdnsService(const std::string &aNetworkName, const std::string &aExtPanId);
+
+    /**
+     * This method checks if mdns service is started.
+     *
+     * @retval true    Started the mdns service.
+     * @retval false   Not started the mdns service.
+     *
+     */
+    bool IsStartedService() { return ot::Mdns::Publisher::GetInstance().IsStarted(); };
```

Comment:
All void functions or methods shall explicitly declare and specify the void type keyword.